src/API/Google/AdsReport.php

```diff
@@ -88,10 +89,17 @@ public function get_report_data( string $type, array $args ): array {
 		} catch ( ApiException $e ) {
 			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
 
-			throw new Exception(
+			$errors = $this->get_api_exception_errors( $e );
+			throw new ExceptionWithResponseData(
 				/* translators: %s Error message */
-				sprintf( __( 'Unable to retrieve campaign report data: %s', 'google-listings-and-ads' ), $e->getBasicMessage() ),
-				$e->getCode()
+				sprintf( __( 'Unable to retrieve report data: %s', 'google-listings-and-ads' ), reset( $errors ) ),
+				$this->map_grpc_code_to_http_status_code( $e ),
+				null,
+				[
+					'errors' => $errors,
+					'type'   => $type,
+					'args'   => $args,
```

Comment:
❓  I take as granted that `$args` doesn't contain any sensitive information that could be accidentally rendered in the Frontend.

Reply:
💅 A bit as the `id` comment... the `type` param was related to the error on my first thought. However, after reading more deeper I noticed it was meant the report Type

Reply:
These are the arguments passed to the request just to help identify what kind of request it was that failed. The arguments would consist of date ranges or fields that need to be displayed in the report (so that would mostly be set by them in the UI).